controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -102,11 +113,57 @@ class ControllerManager : public rclcpp::Node
     std::shared_ptr<controller_interface::ControllerInterface> controller,
     const std::string & controller_name);
 
+  CONTROLLER_MANAGER_PUBLIC
+  controller_interface::ControllerInterface * get_controller_by_name(const std::string & name);
+
+  CONTROLLER_MANAGER_PUBLIC
+  void manage_switch();
+
+  CONTROLLER_MANAGER_PUBLIC
+  void stop_controllers();
+
+  CONTROLLER_MANAGER_PUBLIC
+  void start_controllers();
+
+  CONTROLLER_MANAGER_PUBLIC
+  void start_controllers_asap();
```

Comment:
What does this function do? Is it not every `start_controller` as soon as possible?

Reply:
Right now in ROS2 there's no difference, in ROS1 it was tied to this I believe:
https://github.com/ros-controls/ros_control/blob/noetic-devel/controller_manager_msgs/srv/SwitchController.srv#L13

>start the controllers as soon as their hardware dependencies are ready, will
    wait for all interfaces to be ready otherwise

I haven't dived into the hardware interfaces part yet, so I did not want to make any drastic changes on the behaviour.

Reply:
Ah OK, I wasn't aware of this functionality. Thanks for pointing out!